index.js

```diff
@@ -3,6 +3,39 @@
 // TODO:
 //   - both: support multiline value format (e.g. SOA)
 
+/**
+ * @typedef {object} DnsRecord
+ * @property {string} name The lowercase DNS name without a trailing dot, e.g. `"example.com"`.
+ * @property {number} ttl The TTL in seconds, e.g. `60`.
+ * @property {string} class The DNS class, e.g. `"IN"`.
+ * @property {string} type The record type, e.g. `"A"`.
+ * @property {string} content The record content, e.g. `"2001:db8::1"` or `"example.com."`.
+ * @property {string | null} comment A comment, e.g. `"a comment"`, `null` if absent.
+ */
+
+/**
+ * @typedef {object} DnsData
+ * @property {DnsRecord[]} records Array of `record`
+ * @property {string} [origin] The value of `$ORIGIN` in the zone file.
+ * @property {number} [ttl] The value of `$TTL` in the zone file.
+ * @property {string} [header] An optional header at the start of the file. Can be multiline. Does not include comment markers.
+ */
+
+/**
+ * @typedef {object} ParseOptions
+ * @property {string | null} [replaceOrigin=null] When specified, replaces any `@` in `name` or `content` with it.
+ * @property {boolean} [crlf=false] When true, emit `\r\n` instead of `\n` in `header`.
+ * @property {number} [defaultTTL=60] Default TTL when absent and `$TTL` is not present.
+ * @property {boolean} [dots=false] Ensure trailing dots on FQDNs in content. Supports a limited amount of record types.
+ */
+
+/**
+ * @typedef {object} StringifyOptions
+ * @property {boolean} [sections=true] Whether to group records into sections.
+ * @property {boolean} [crlf=false] When `true`, emit `\r\n` instead of `\n` for the resulting zone file.
+ * @property {boolean} [dots=false] Ensure trailing dots on FQDNs in content. Supports a limited amount of record types. Default: `false`.
+ */
+
```

Comment:
Is there an option to keep those docstrings out of the source? I prefer not to have them there.

Reply:
These structure definition is reference in the following jsdocs, remove them will break the jsdoc type annotation.